gcloud/logging/logging_service_v2_api.py

```diff
@@ -0,0 +1,245 @@
+        The log will reappear if it receives new entries.
+
+        """
+
+        delete_log_request = logging_pb2.DeleteLogRequest(
+            log_name=log_name,
+            **kwargs)
+        return self.delete_log_callable()(delete_log_request)
+
+    def delete_log_callable(
+            self,
+            is_retrying=None,
+            max_attempts=None):
+        return api_callable.idempotent_callable(
+            self.stub.DeleteLog,
+            is_retrying=is_retrying,
+            max_attempts=max_attempts,
+            defaults=self.defaults)
+
+    def write_log_entries(
+            self,
+            log_name='',
+            resource=None,
+            labels=None,
+            entries=None,
+            **kwargs):
+        """
+        Writes log entries to Cloud Logging.
+        All log entries in Cloud Logging are written by this method.
+
+        """
+        if resource is None:
+            resource = monitored_resource_pb2.MonitoredResource()
+        if labels is None:
+            labels = []
+        if entries is None:
+            entries = []
+        write_log_entries_request = logging_pb2.WriteLogEntriesRequest(
+            log_name=log_name,
+            resource=resource,
+            labels=labels,
+            entries=entries,
+            **kwargs)
+        return self.write_log_entries_callable()(write_log_entries_request)
+
+    def write_log_entries_callable(
+            self,
+            is_retrying=None,
+            max_attempts=None):
+        return api_callable.idempotent_callable(
+            self.stub.WriteLogEntries,
+            is_retrying=is_retrying,
+            max_attempts=max_attempts,
+            defaults=self.defaults)
+
+    def list_log_entries(
+            self,
+            project_ids=None,
+            filter='',
+            order_by='',
+            **kwargs):
+        """
+        Lists log entries.
+        Use this method to examine log entries from Cloud Logging.
+        See [Exporting Logs](/logging/docs/export) for other ways to copy
+        log entries out of Cloud Logging.
+
+        """
+        if project_ids is None:
+            project_ids = []
+        list_log_entries_request = logging_pb2.ListLogEntriesRequest(
+            project_ids=project_ids,
+            filter=filter,
+            order_by=order_by,
+            **kwargs)
+        return self.list_log_entries_callable()(list_log_entries_request)
+
+    def list_log_entries_callable(
+            self,
+            is_retrying=None,
+            max_attempts=None,
+            page_streaming=_LIST_LOG_ENTRIES_DESCRIPTOR):
+        return api_callable.idempotent_callable(
+            self.stub.ListLogEntries,
+            page_streaming=page_streaming,
+            is_retrying=is_retrying,
+            max_attempts=max_attempts,
+            defaults=self.defaults)
+
+    def read_log_entries(
+            self,
+            project_ids=None,
+            filter='',
```

Comment:
Suggest using `filter_` instead of `filter`.

In general you can check if a string is a built-in via

```python
>>> 'filter' in dir(__builtins__)
True
```